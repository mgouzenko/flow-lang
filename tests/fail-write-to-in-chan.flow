// Should not be able to write to an in channel

proc numGen(out int ochan){
  list <int> test = [1, 2, 3, 4, 5];

  while(#test > 0) {
    @test -> ochan;
    test = ^test;
  }
  poison ochan;
}

proc sum(in int chan) {
  while(chan) {
    3 -> chan;  // FAIL
    print_int(@chan);
  }
}

int main() {
  channel<int> chan;
  numGen(chan);
  sum(chan);
}