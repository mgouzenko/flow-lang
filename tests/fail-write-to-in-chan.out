Fatal error: exception Invalid_argument("Invalid write to channel")